### PR TITLE
refactor(communities): sort categories and chats

### PR DESCRIPTION
**Pull request by richard-ramos**
- Fixes: #4543
- Fixes: #4521

### CODE REVIEW

File: src/app/modules/main/communities/controller.nim [hunk 1]
```diff
@@ -44,6 +44,10 @@ method init*(self: Controller) =
 method getAllCommunities*(self: Controller): seq[CommunityDto] =
   result = self.communityService.getAllCommunities()
 
+  self.events.on(SIGNAL_COMMUNITY_CATEGORY_REORDERED) do(e:Args):
```
Comment (Khushboo-dev-cpp) [inline src/app/modules/main/communities/controller.nim:47]: remove as this doesn't nothing?

Comment (Khushboo-dev-cpp) [inline src/app/modules/main/communities/controller.nim:47]: remove this as it is unused? also maybe emitting signal is not needed? handled in chat_sections controller

**Review by Khushboo-dev-cpp — CHANGES_REQUESTED** [review]
I just found one small leftover code

**Comment by anastasiyaig** [comment]
sometimes reordering channels does not make any effect, not sure the reason @richard-ramos could u please also check?


https://user-images.githubusercontent.com/82375995/152168213-84c929cd-be39-46f4-ba18-f5200a687632.mov

**Comment by anastasiyaig** [comment]
@richard-ramos  will u have time to look into this?

**Comment by richard-ramos** [comment]
I'll have a look now

**Comment by richard-ramos** [comment]
@anastasiyaig latest commit in this branch should fix the issue

**Review by Khushboo-dev-cpp — APPROVED** [review]
thank you :)

**Review by anastasiyaig — APPROVED** [review]

**Review by iurimatias — APPROVED** [review]

**Pull request merged**